web-bluetooth/service-worker.js

```diff
@@ -1,20 +1,27 @@
-self.addEventListener('fetch', function(event) {
-  const request = event.request;
-  const url = new URL(event.request.url);
+function addToCache(request, networkResponse) {
+  return caches.open('web-bluetooth')
+    .then((cache) => cache.put(request, networkResponse.clone()));
+}
 
-  // Don't cache anything that is not on this origin.
-  if (url.origin !== location.origin) {
-    return;
-  }
+function getCacheResponse(request) {
+  return caches.open('web-bluetooth').then((cache) => {
+    return cache.match(request);
+  });
+}
 
-  event.respondWith(caches.open('web-bluetooth').then(cache => {
-    return cache.match(request).then(response => {
-      var fetchPromise = fetch(request).then(networkResponse => {
-        cache.put(request, networkResponse.clone());
-        return networkResponse;
+function getNetworkOrCacheResponse(request) {
+  return new Promise((resolve) => {
```

Comment:
See https://twitter.com/domenic/status/765955660026707969 for a discussion as to why using `new Promise()` like that is an anti-pattern. (I assume this code was carried over from @philipwalton's original post, as he's since updated it.)